matrix-sdk/src/main/java/org/matrix/androidsdk/crypto/MXCrypto.java

```diff
@@ -560,10 +572,12 @@ public void run() {
                         mOlmDevice = null;
                     }
 
-                    mMyDevice = null;
+                    // Do not reset My Device
```

Comment:
Unitary tests may fail because of this.
There is no need to reset those two members, because if it is accessed later (because of an asynchronous callback for instance), it will crash.
The Garbage collector won't be blocked because of not resetting to null those two members.